src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java

```diff
@@ -515,6 +518,10 @@ public void setupMetadata(FeatureTypeInfo ftinfo, FeatureSource featureSource)
     public ReferencedEnvelope getLatLonBounds(ReferencedEnvelope nativeBounds,
             CoordinateReferenceSystem declaredCRS) throws IOException {
         if (nativeBounds != null && declaredCRS != null) {
+            // shortcut: return empty envelope if nativeBounds is empty
+            if (nativeBounds.isEmpty()) {
+                return new ReferencedEnvelope(nativeBounds, DefaultGeographicCRS.WGS84);
```

Comment:
You should check the WFS spec and see if it has an explicit notion of empty envelope, if not, it must be a non empty one.